packages/core/http/core-http-router-server-internal/src/versioned_router/core_versioned_route.ts

```diff
@@ -161,19 +161,19 @@ export class CoreVersionedRoute implements VersionedRoute {
     ) {
       try {
         const { body, params, query } = validate(req, validation.request, handler.options.version);
-        req.body = body;
-        req.params = params;
-        req.query = query;
+        (req as Mutable<KibanaRequest>).body = body;
```

Comment:
I thought the cast at line 120 was precisely to avoid this 😮

Reply:
On line 134 we have a type guard that changes the type to `KibanaRequestWithQueryVersion` and then we mutate `req` as that type. Seems like typescript remembers this so later in this part of the code the type becomes `Mutable<KibanaRequest> | KibanaRequestWithQueryVersion`

We can clean this up a bit by changing the type guard to `Mutable< KibanaRequestWithQueryVersion>` I can submit a PR

Reply:
Thanks for pointing this out, I was also surprised. Rudolf just pointed to the following https://github.com/elastic/kibana/pull/162738#pullrequestreview-1565270373

Reply:
https://github.com/mistic/kibana/pull/17

Reply:
👍